django_lti_tool_provider/views.py

```diff
@@ -0,0 +1,148 @@
+from django.views.generic import View
+import oauth2
+import logging
+
+from django.conf import settings
+from django.core.urlresolvers import reverse, NoReverseMatch
+from django.http import HttpResponseRedirect, HttpResponseBadRequest
+from django.views.decorators.csrf import csrf_exempt
+
+from ims_lti_py.tool_provider import DjangoToolProvider
+
+from django_lti_tool_provider.models import LtiUserData
+from django_lti_tool_provider.signals import Signals
+
+
+_logger = logging.getLogger(__name__)
+
+
+class LTIView(View):
+    authentication_hooks = []
+
+    PASS_TO_AUTHENTICATION_HOOK = {
+        'lis_person_sourcedid': 'username',
+        'lis_person_contact_email_primary': 'email',
+        'user_id': 'user_id'
+    }
+
+    @csrf_exempt
+    def dispatch(self, *args, **kwargs):
+        return super(LTIView, self).dispatch(*args, **kwargs)
+
+    def get(self, request, *args, **kwargs):
+        return self.process_request(request)
+
+    def post(self, request, *args, **kwargs):
+        return self.process_request(request)
+
+    def process_request(self, request):
+        if not request.user.is_authenticated():
+            try:
+                lti_parameters = self._get_lti_parameters_from_request(request)
+                lti_data = {
+                    hook_name: lti_parameters.get(lti_name, None)
+                    for lti_name, hook_name in self.PASS_TO_AUTHENTICATION_HOOK.items()
+                }
+            except oauth2.Error, e:
+                _logger.exception(u"Invalid LTI Request")
+                return HttpResponseBadRequest("Invalid LTI Request: " + e.message)
+
+            for hook in self.authentication_hooks:
+                hook(request, lti_data)
+
+        if request.user.is_authenticated():
+            return self.process_authenticated_lti(request)
+        else:
+            return self.process_anonymous_lti(request)
+
+    @classmethod
```

Comment:
@smarnach because it doesn't use any instance methods/variables; and some linters (e.g. PyCharm) suggest converting to classmethod/staticmethod.